app/com/linkedin/drelephant/tuning/AutoTuningAPIHelper.java

```diff
@@ -441,19 +441,22 @@ private JobExecution getJobExecution(TuningInput tuningInput) {
       JobExecution jobExecution) {
     JobSuggestedParamSet jobSuggestedParamSet;
     logger.debug("Finding parameter suggestion for job: " + jobExecution.job.jobName);
+    List<JobSuggestedParamValue> jobSuggestedParamValues = null;
     if (tuningInput.getRetry()) {
-      logger.info(" Retry ");
+      logger.info(" Retried job execution " + tuningInput.getJobExecId());
```

Comment:
Is this method called when execution is re-tried or is about to retry?

Reply:
When an execution is retried, a dummy getCurrentRunParameters call is made to dr. elephant by setting "IS_RETRY_API_PARAM" to true. In this way dr. elephant will know that there was a failure with earlier parameters. And if "IS_RETRY_API_PARAM" is true then tuningInput.getRetry() evaluates to true. So this method is called when execution is re-tried.

Reply:
Thanks.